File: python/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        l, r, max_profit = 0, 1, 0

        while r < len(prices):
            if prices[r] > prices[l]:
                profit = prices[r] - prices[l]
                max_profit = max(profit, max_profit)
            else:
                l = r
            r += 1

        return max_profit

File: javascript/0121-best-time-to-buy-and-sell-stock.js
var maxProfit = function(prices) {
    let [l, r, maxProfit] = [0, 1, 0]
    while (r < prices.length) {
        if (prices[r] > prices[l]) {
            profit = prices[r] - prices[l]
            maxProfit = Math.max(profit, maxProfit)
        }
        else {
            l = r
        }
        r++
    }
    return maxProfit
};
